### PR TITLE
docs: add ecosystem release info

### PR DESCRIPTION
**Pull request by mattrosno**
**Until this PR is reviewed and merged, the current name of Carbon's upcoming major release is the "2021 Release", as stated in the [Medium article](https://medium.com/carbondesign/whats-coming-to-carbon-in-2021-39a4c7d1762a).**

There is currently a concern from a member of Carbon's advisory board that calling Carbon's next major release "v11" will cause unnecessary confusion from offering teams thinking that because Carbon 10 required a product redesign, the next major version will as well.

I believe that concern can be mitigated by clearly stating how Carbon 10 was unique due to the new version of IBM Design Language in 2019, and stating how the next major version remains on that same IBM Design Language version so another product redesign is not necessary.

There are two parts to this pull request.

# 1) Release table

The goal of this information it to state how we have system releases, that contain independently-versioned package releases. That's always been a cause of confusion (e.g. "wait, Carbon 10 is v7 for React components!?").

# 2) Release naming

This pull requests states `Carbon TBD` for now in the updated `/releases` page. Options that are being discussed in this PR:

* Carbon 10 [Codename]
* Carbon 10 [Version]
* Carbon 11
* Carbon 2021
* Carbon [Codename]

# Changelog

**New**

- Table that shows system releases and package releases, in the changelog and roadmap page

**Removed**

- Resource tiles to 2020Q4 OKRs

### CODE REVIEW

**Comment by mattrosno** [comment]
We should also link to release radar so people understand we ship weekly minors, and not once every couple years.

https://github.com/carbon-design-system/carbon/wiki/Release-radar

**Comment by alisonjoseph** [comment]
Thanks for putting this together. Should the title/name of the page change at all to reflect that this is where release information can be found? I'm a little worried it's "hidden" on the Changelog and roadmap page. Maybe @janchild can chime in? 
Do we need/want to put release number information in the chart around the community-supported frameworks?

Agree we should add a paragraph below the chart maybe around minor releases and have a resource card? link out release radar.

**Comment by janchild** [comment]
This looks great, @mattrosno. I like it. Thanks for the heads up, @alisonjoseph. 

I would change the title of the page to "Releases"—it's nice and general and will cover the changelog and roadmap resources also. And then I'd change that second section to "Release schedule". That's what it is. It's nice to go back with the historical data, really nice.

File: src/pages/whats-happening/changelog-and-roadmap/index.mdx
Comment (tay1orjones) [inline src/pages/whats-happening/changelog-and-roadmap/index.mdx:54]: Could we place this under a heading? It could be useful to be able to bookmark/link directly to this text in our support channels - slack, GH issues/discussions, etc.

```suggestion
### Carbon 11

Where the Carbon team strives for frictionless major version updates through
```

**Review by tay1orjones — COMMENTED** [review]
+1 for "Carbon 11". The content looks great to me - very clear and concise 👍 

I agree this feels like a communication problem more-so than a naming problem. Have we received direct feedback or questions highlighting redesign/IDL versioning as an assumption or point of confusion?

Also as we're approaching alpha, we could reinstate the BLM/Voting alert banner removed in https://github.com/carbon-design-system/carbon-website/pull/1968 to call this out and link to this page. There are lots of different ways we could phrase it

- > "Carbon is evolving with v11.0.0!  [Learn more about the upcoming release]()" 
- > "Carbon is evolving with v11!  [And it won't require a redesign]()" 
- > "Carbon is evolving with v11!  [No redesign required]()" 
- > "Carbon 11 is approaching. [Learn how to prepare your product]()"
- > "A new major version release is in the works! [Based on the same version of IBM Design Language]()"
- > "Carbon 11 is coming soon! [Learn what it means for your team]()"

**Comment by mattrosno** [comment]
>Should the title/name of the page change at all to reflect that this is where release information can be found?

Done.

>And then I'd change that second section to "Release schedule".

Done.

>Agree we should add a paragraph below the chart maybe around minor releases and have a resource card? link out release radar.

Done.

>Could we place this under a heading? It could be useful to be able to bookmark/link directly to this text in our support channels - slack, GH issues/discussions, etc.

Done.

**Comment by mattrosno** [comment]
Reasons "10" should not be in the name of the system's next major release:

1. Our users, package managers, and the industry expect increments in major versions to represent breaking changes
2. Our next major release will contain breaking changes, so none of the libraries will carry the "10" major version
3. Our community had no problems understanding the 6 -> 7, 7 -> 8, 8 -> 9 migrations, all of which were IDL 1 and did not require product redesigns
4. We can mitigate false assumptions of Carbon 11 requiring product redesigns by clearly stating it in this new page and including this primary message in our release comms (e.g. Slack messages, Medium articles, website banners)
5. We can include a call to action in release comms (e.g. questions on the impact of this release? Contact Matt.)

**Comment by chrisconnors-ibm** [comment]
I can confirm plugin teams throughout Cloud are _extremely_ antsy about anything that refers to “Carbon 11”. 

“You just put us through 2 years of effort to migrate to carbon 10 and now you expect us to do that again?” is the feedback from OM, DEV leadership, and above. "Why is design always pulling the rug out from under us" is a direct quote.

Carbon 10 (the brand) has nothing to do with the semantically versioned packages  themselves (which aren't aligned at all, carbon @ 10.32.1, carbon-react @ 7.2, etc.).

Carbon 10 is a brand. If you change that, you're going to exert a lot of effort to _try_ and convince people nothing has changed or the adoption lift is not that heavy, effort the Carbon team could better apply elsewhere. You're also going to spend a lot of trust capital with the teams who rely on this work.

If you align _all_ the package versions to all start with 10, and then use the second value for "major" (annual) releases, followed by minor, it's unlikely we hear any of these objections at the organizational levels we're hearing them from.

brand.major.minor
10.40.0, 10.40.1, 10.40.2… 10.40.n

**Comment by joshblack** [comment]
Unfortunately, I think this problem is the perfect blend of brand, technical, PR...everything lol. Much of the challenge seems to be around tying Carbon to the IDL when what is proposed suggests that they are independent. It seems like it'd be helpful to think of them as separate because of the nature of changes that occur when a DL changes versus a design system changes.

Related to that, there would be technical hiccups for why the version ideas proposed won't work for keeping things at 10.x. Unfortunately, we run into:

- Semantic Versions only include 3 numbers, any additional numbers are dropped (for the idea of a four digit version system)
- We've already used tags in the 10.x range for our minor releases. This would make it awkward to do a `10.1` release when we already have a 10.1 tag from 2 years back and are currently on 10.32
  - For a full list of versions, here's the [release radar wiki](https://github.com/carbon-design-system/carbon/wiki/Release-radar)

The rationale behind the second bullet point is similar to how other platforms/OS operate. You have the platform/OS version and then applications have versions themselves. Applications also target specific platform/OS ranges. For example, macOS 11.2 has Safari 14, Finder 11.1, Terminal 2.11, etc. This also indicates that packages will need the flexibility to move beyond the "leading" version, in other words Safari 14 > macOS 11.2.

When it comes to Carbon 10 as a brand, this feels like a distinction made outside of our group. IDL was/is the brand, in my opinion. Our system should/will have reasons to have major changes distinct from that brand. It's surprising hearing that feedback from Cloud who has been with the project through major versions that targeted the same brand.

If anything else, it may be helpful to move forward with 11 as the name to make this distinction clear to consumers that major versions of a design system will not mean a brand update. It also would encode that major versions of a design system still carry breaking changes, but of a different nature than brand updates where you need to redesign a product.

Just my 2c, curious to hear/learn more in this space 👀

**Comment by mattrosno** [comment]
>If you align all the package versions to all start with 10, and then use the second value for "major" (annual) releases, followed by minor, it's unlikely we hear any of these objections at the organizational levels we're hearing them from.

@chrisconnors-ibm unfortunately, that is not how the industry-standard semantic versioning specification works. Package managers (tooling) and users alike expect the first number to represent incompatible API changes.

>“You just put us through 2 years of effort to migrate to carbon 10 and now you expect us to do that again?” is the feedback from OM, DEV leadership, and above. "Why is design always pulling the rug out from under us" is a direct quote.

@chrisconnors-ibm thank you for passing that along. When that happens, can you include us in your response so we can have a better pulse on such questions and can respond accordingly in our communications?

**Comment by chrisconnors-ibm** [comment]
> It's surprising hearing that feedback from Cloud who has been with the project through major versions that targeted the same brand.

That's because GMs and OM Leads and VPs probably aren't in the audience for your outreach to date. CloudPAL was not alarmed by a 10-11 change, but their constituents very much are.

**Comment by mattrosno** [comment]
The intent of Carbon 10 was never to a brand, but the initial means to use the new IBM Design Language (the brand). Over the last two years, was the "Carbon 10" name was over-indexed through migration mandates so "Carbon 10" became synonymous with product redesigns to express IBM's new brand?

If so, that's not good, but I believe that can be corrected through our release communication strategy, so we can remain true to semantic versioning (both our ecosystem release naming and the naming of each package). In the long run, this is best for our users.

If "Carbon 10" is now unintentionally synonymous to the current IDL, if we don't believe that we can correct how "Carbon 10" has been used the last two years (meaning we can't correct how "Carbon 11" may be perceived), then an option would be to drop numbered ecosystem releases all together and start fresh. To set new expectations with a clean slate.

One example is Rust (e.g. [Rust 2018](https://doc.rust-lang.org/edition-guide/rust-2018/index.html)) or ECMAScript (e.g. [ES 2020](https://en.wikipedia.org/wiki/ECMAScript#11th_Edition_%E2%80%93_ECMAScript_2020)). Since we're more or less shipping time-based annual major releases, our next ecosystem major release could be named "Carbon 2021", with our packages continuing to follow semvar, and we continue to set expectations on what each release means.

"Carbon 11" is ideal. I can see how we could effectively run with "Carbon 2021". I do not see the value in keeping the "Carbon 10" ecosystem name when that no longer tracks the primary styling package. That will just add more confusion.

**Comment by aagonzales** [comment]
I don't think we should say the exact term "Carbon 11." I think that's the true worrying brand point of the one advisory board member, that we've moved past "Carbon 10" which has a lot associations with it whether true or not. We should be saying "Carbon Version 11" implying its not Carbon the brand that's changing. 

I think it would help for us to communicate the distinctions between Brand, IBM Design Language and Carbon. I also think we need to explain how versioning works for each of those things and how the different versions relate to each other. I think we've been assuming our users know all things and the differences between them which is a very big and probably wrong assumption. I think this is likely the route of all this confusion.

----

Also for the record, I don't think having a non-numbered _release_ name is as concerning to me as other members of the team. I don't think we'll be having more than one major release a year if that so i think the argument on needing sequential naming is less strong, though I do see the benefit of it, its just not a hill I would choose to die on. We only support one version back so really its only important to know about the current release and the one before that. 

Our code base will always be properly versioned so we won't be breaking with standards in the sense. 

I actually think since many of the packages in the system wont be at v11 (since they have their own versioning) that a non-version name for the release could help unify across the ecosystem. 

**Magenta Release**
Release date: Q2 2021
Carbon version: v11.0.0
Elements package: v11.0.0
React package: v8.0.0
Vanilla package: N/A
Angular package: v4.0.0
Cloud PAL: v2.0.0

**Comment by mattrosno** [comment]
>I also think we need to explain how versioning works for each of those things and how the different versions relate to each other.

@aagonzales I'm hoping this new website page will better communicate that. Any suggestions based on what's in the updated `/releases` page so far?

**Comment by alisonjoseph** [comment]
> Also for the record, I don't think having a non-numbered release name is as concerning to me as other members of the team. I don't think we'll be having more than one major release a year if that so i think the argument on needing sequential naming is less strong, though I do see the benefit of it, its just not a hill I would choose to die on.

I would agree with this. As long as our package numbers stay true to semver then what we call it is less important.

**Comment by aagonzales** [comment]
oh oops guess that's why i should look at the preview and not just the files changed tab. Had to see that in that context. I really like that table, I think it explain Carbon's relationship to those things but I dont think it communicates the brand versioning well. Do people know what IDL v2 means? Because in the beginning we called it "duo". Maybe more recent employee don't know there was a v1.  

 Like IDL needs to talk about its versioning somewhere and we need to help people understand that Carbon isn't a style, its the system. I think people that don't know Carbon's history (basically anyone outside of Cloud) might not understand that.

File: src/pages/whats-happening/releases/index.mdx
Comment (aagonzales) [inline src/pages/whats-happening/releases/index.mdx:63]: I feel like we're only talking about design here. I think inline theming could actually be a big effort for people to transition to from existing implementation (or its a know concern). Also we're changing our entire color token system. Neither are as big as fully branding redesign but

**Comment by joshblack** [comment]
@chrisconnors-ibm 

> That's because GMs and OM Leads and VPs probably aren't in the audience for your outreach to date. CloudPAL was not alarmed by a 10-11 change, but their constituents very much are.

The distinction I was trying to make was that I don't think teams who have been with Carbon would believe a breaking change carries a DL change. They would 100% be worried about a major change because it means having to spend cycles away from product work towards addressing the breaking changes that impact them. I think this is something that Cloud PAL 2.0 will be impacted by as well as us, and it's totally understandable from the lenses of a product team.

The best part about the release since it's not a visual change will be that they don't have to migrate right away. If what they have is working, great. Keep it that way.

Happy to engage directly with GMs, OMs, VPs to clarify this stuff. Is there a good format in which we could communicate this outside of Medium? Would this be something that we should publish on W3 Publisher?

**Comment by mattrosno** [comment]
Thanks for the discussion today. I pushed an update so for now, the new releases page states `Carbon TBD` for the upcoming major release. The goal of this page is to be landing page that we can link to from any Slack announcement, Medium article, etc. to explain the major releases in detail and in relation to each other. @janchild I made another pass at that last paragraph. I also put IDL as the first column in the table because brand drives how the design system digitally expresses the brand.

https://carbondesignsy-git-fork-mattrosno-release-naming-carbon-840e6a.vercel.app/whats-happening/releases/

I also updated the PR description to state release name options so far, as naming considerations have been shared in PR comments. (CC @mjabbink)

File: src/pages/whats-happening/releases/index.mdx
Comment (mattrosno) [inline src/pages/whats-happening/releases/index.mdx:67]: From @mjabbink:

something along these lines

(from)

Carbon v10 is the digital expression of the IBM Design Language, v2. The updated components required some product redesigns, however they had a clearly identifiable IBMness and created a connective tissue between all product experiences.

(to)

Carbon v10 was the release that implemented our newly updated IBM brand expression (Design Language v2) into our components. The updated components required product teams to redesign product UIs and begin a plan to migrate and/or adopt the new components to copy to IBM brand experience standards.

**Comment by mjabbink** [comment]
Just noticing the rules are not hanging and the type should align left. @jeanservaas also wonder if this needs white background? I also thought we had an issue to darken the rules. They are barely visible on G10.

**Comment by mattrosno** [comment]
https://carbondesignsy-git-fork-mattrosno-release-naming-carbon-840e6a.vercel.app/whats-happening/releases/

@janchild I updated to Carbon v11 and added in an article card, so we can put the new Medium article card there too once it's published. Can you review and let me know what you want for the page?

@alisonjoseph @joshblack it'd be nice to include community packages like Carbon Angular, Vue, etc. in the table, but if we go that route, we'll keep to come up with a better way to present all that data as the current MDX table isn't meant for that many columns. Should we defer that for now and keep this table to the packages where we are primary maintainers?

@tw15egan @aagonzales thoughts on the column 1 bullet points? Do we want to populate older releases too or is that just water under the bridge (let's keep focus on v10, v11, and the impact of each)?

**Review by tw15egan — APPROVED** [review]

**Review by vpicone — COMMENTED** [review]
Rewrite rules look okay

**Review by vpicone — APPROVED** [review]

**Review by vpicone — CHANGES_REQUESTED** [review]
Broken on mobile. image

**Comment by mjabbink** [comment]
@mattrosno I think you can remove “visuals” 

Screen Shot 2021-04-19 at 6 13 56 PM

**Comment by alisonjoseph** [comment]
I think it's fine for now to leave it to the main packages we maintain. Would be nice for the future to figure out a way to display a historical list for all packages.

**Comment by alisonjoseph** [comment]
@vpicone why did I think our markdown tables auto scrolled in mobile? 🤔

**Comment by vpicone** [comment]
> @vpicone why did I think our markdown tables auto scrolled in mobile? 🤔 

They do, @mattrosno you might consider using [markdown table](https://gatsby-theme-carbon.vercel.app/components/markdown#tables) from the theme instead of the StructuredList component manually here.

File: src/pages/whats-happening/releases/index.mdx
Comment (janchild) [inline src/pages/whats-happening/releases/index.mdx:103]:
```suggestion
          — Q2 2021
```

File: src/pages/whats-happening/releases/index.mdx
Comment (janchild) [inline src/pages/whats-happening/releases/index.mdx:124]:
```suggestion
          — 29 Mar 2019
```

File: src/pages/whats-happening/releases/index.mdx
Comment (janchild) [inline src/pages/whats-happening/releases/index.mdx:156]:
```suggestion
        <UnorderedList>— 4 June 2018</UnorderedList>
```

File: src/pages/whats-happening/releases/index.mdx
Comment (janchild) [inline src/pages/whats-happening/releases/index.mdx:176]:
```suggestion
        <UnorderedList>— 26 Oct 2017</UnorderedList>
```

File: src/pages/whats-happening/releases/index.mdx
Comment (janchild) [inline src/pages/whats-happening/releases/index.mdx:196]:
```suggestion
        <UnorderedList>— 30 Mar 2017</UnorderedList>
```

**Review by janchild — COMMENTED** [review]
@matt, you can remove the Medium article card as we discussed today. And here is the new para for v11:

Carbon v11 focuses on quality of life updates for designers and developers including token and prop renaming for ease of use, component API updates for a more predictable developer experience, and introducing CSS grid for accurate layout building in line with the design language. With no changes to the IBM Design Language, v11 will not require any brand-driven product redesigns and teams are free to update at their convenience.

**Comment by janchild** [comment]
> @mattrosno I think you can remove “visuals.” Design Language are visual elements. So it reads a bit like “Updates visuals to new visuals” to me.
> 
> Screen Shot 2021-04-19 at 6 13 56 PM

We could put 
- Updates components to new design language

File: src/pages/whats-happening/releases/index.mdx
Comment (janchild) [inline src/pages/whats-happening/releases/index.mdx:126]:
```suggestion
          — Updates components to new design language
```

File: src/pages/whats-happening/releases/index.mdx
Comment (janchild) [reply src/pages/whats-happening/releases/index.mdx:81]: @mattrosno Suggested tweak to this: 

Carbon v10 implemented our newly updated IBM brand expression (Design Language v2) into our components. The updated components required product teams to redesign product UIs and begin a plan to migrate and/or adopt the new components to comply with IBM brand experience standards.

**Comment by mjabbink** [comment]
update 

FROM — Updates visuals to new design language  TO  — Updates to new IBM design language

File: src/pages/whats-happening/releases/index.mdx
Comment (mattrosno) [reply src/pages/whats-happening/releases/index.mdx]: @aagonzales we'll be able to talk specifics like inline theming in a new page that Jan, Josh, and I have discussed with that Carbon v11 FAQ content.

File: src/pages/whats-happening/releases/index.mdx
Comment (mattrosno) [reply src/pages/whats-happening/releases/index.mdx:81]: Done!

**Comment by mattrosno** [comment]
@janchild ready for review. https://carbondesignsy-git-fork-mattrosno-release-naming-carbon-840e6a.vercel.app/whats-happening/releases/

File: src/pages/whats-happening/releases/index.mdx
Comment (janchild) [inline src/pages/whats-happening/releases/index.mdx:20]:
```suggestion
Learn about our latest updates, our release history, and what the team has planned for 
```

File: src/pages/whats-happening/releases/index.mdx
Comment (janchild) [inline src/pages/whats-happening/releases/index.mdx:4]:
```suggestion
Learn about our latest updates, our release history, and what the team has planned 
```

**Review by janchild — COMMENTED** [review]
Just a few copyedits in here, Matt! Let me know if you have any questions.

**Review by janchild — APPROVED** [review]
🎉

File: src/pages/whats-happening/releases/index.mdx
Comment (vpicone) [inline src/pages/whats-happening/releases/index.mdx:80]: Is there a reason for the wrapper divs here? If you need to target them with css, a span with a classname would be more appropriate.

Comment (mattrosno) [reply src/pages/whats-happening/releases/index.mdx:80]: @vpicone I need line breaks. Ideally, unordered list elements, but those render as 16px font size which is bigger than the markdown table font size, so that's not an option right now. I previously had `<br />` elements in there, but Prettier formatting was busting the table. Then I switched it to divs and Prettier still busted the table. Then I added the Prettier ignore statements and called it good.

Comment (vpicone) [reply src/pages/whats-happening/releases/index.mdx:80]: Right on. If you're ignoring it anyway, can we go back to break elements to get the line breaks?

I just don't want the next person editing the table to be confused with the empty wrappers.

Comment (mattrosno) [reply src/pages/whats-happening/releases/index.mdx:80]: @vpicone will do, I'll swap back to line break elements.

**Review by vpicone — APPROVED** [review]
Good to go once the divs are yoinked

**Review by emyarod — APPROVED** [review]
the markdown table seems to render properly with line breaks and no prettier ignore flags

**Pull request merged**
